File: vipclaw-agent/vipclaw-agent-service/src/main/java/com/vipamp/vipclaw/agent/service/dto/ConfirmRequest.java
package com.vipamp.vipclaw.agent.service.dto;

import java.util.List;

/**
 * @Author: heqingsong
 * @Date: 2026/4/12
 * @Description: ChatRequest
 * @Project: vipclaw
 */
public record ConfirmRequest(String sessionId,
                             boolean isConfirmed,
                             List<ToolInfo> toolInfoList,
                             Boolean enableThink,
                             Boolean enableSearch) {

    public record ToolInfo(String toolId, String toolName) {

    }
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/ascopagent/dto/ConfirmRequest.kt
package com.vipamp.vipclaw.ascopagent.dto

import io.swagger.v3.oas.annotations.media.Schema

/**
 * Confirm 请求 DTO
 *
 * @Author: heqingsong
 * @Date: 2026/4/12
 * @Project: vipclaw
 */
@Schema(description = "Confirm 请求")
data class ConfirmRequest(
    @Schema(description = "会话 ID")
    val sessionId: String,

    @Schema(description = "是否确认")
    val isConfirmed: Boolean = false,

    @Schema(description = "工具信息列表")
    val toolInfoList: List<ToolInfo> = listOf(),

    @Schema(description = "是否启用思考模式")
    val enableThink: Boolean = false,

    @Schema(description = "是否启用搜索")
    val enableSearch: Boolean = false
) {
    /**
     * 工具信息
     */
    @Schema(description = "工具信息")
    data class ToolInfo(
        @Schema(description = "工具 ID")
        val toolId: String? = null,

        @Schema(description = "工具名称")
        val toolName: String? = null
    )
}
